onadata/apps/api/storage.py

```diff
@@ -0,0 +1,64 @@
+"""
+Backends module for th API app
+"""
+import logging
+
+from django.conf import settings
+from django.utils import timezone
+
+from django_digest.backend.storage import AccountStorage
+from django_digest.models import PartialDigest
+
+from onadata.apps.api.models.odk_token import ODKToken
+
+_l = logging.getLogger(__name__)
+_l.setLevel(logging.DEBUG)
+
+ODK_KEY_LIFETIME_IN_SEC = getattr(settings, 'ODK_KEY_LIFETIME', 7) * 86400
+
+
+def _check_odk_token_expiry(created):
+    """
+    Checks to see whether an ODK Token has expired
+    """
+    time_diff = (timezone.now() - created).total_seconds
+
+    return time_diff > ODK_KEY_LIFETIME_IN_SEC
+
+
+class DigestAccountStorage(AccountStorage):
```

Comment:
I would probably propose a more aligned naming  `ODKTokenAccountStorage`. `Digest` is too generic and is not as specific to the class referred here. Or perhaps the functionality should be adapted to handle both cases as we had discussed.

Reply:
Changed in latest commits